welt/attention.py

```diff
@@ -22,7 +35,7 @@ def add_self_attention_blocks(mask: torch.Tensor, words: list[str]) -> None:
                     "Skipping self-attention block.",
                     stacklevel=2)
             else:
-                mask[0, shift_out_idx:shift_out_idx + i, shift_out_idx:shift_out_idx + i] = 1
+                blocks.append((shift_out_idx, i))
```

Comment:
use `yield` rather than array of `blocks`.
add a test in test_attention specifically for this getting the correct indexes